llvm/lib/Target/AIE/AIEBaseInstrInfo.h

```diff
@@ -29,42 +29,61 @@ namespace llvm {
   using TargetInstrInfo::TargetInstrInfo;
-  /// Return the opcode for a return instruction
 
-  virtual unsigned getReturnOpcode() const = 0;
+  /// Return the opcode for a return instruction
+  virtual unsigned getReturnOpcode() const {
+    llvm_unreachable("Target didn't implement getReturnOpcode");
+  }
 
   /// Return the opcode for a call instruction
   /// \param CallerF The function that makes the call
   /// \param IsIndirect Select function pointer call or direct call
   /// \param Select a tail call variant.
   virtual unsigned getCallOpcode(const MachineFunction &CallerF,
-                                 bool IsIndirect, bool IsTailCall) const = 0;
+                                 bool IsIndirect, bool IsTailCall) const {
+    llvm_unreachable("Target didn't implement getCallOpcode");
+  }
+
   /// Return the kind of slot that this instruction can be executed in.
   /// This is used to direct the packetization of simple instructions.
   /// NOTE: If this is called on a Composite Instruction (i.e. an instruction
   /// defining a Packet format, owning possibly multiples slots), the returned
   /// slot will be the default one (unknown).
-  virtual MCSlotKind getSlotKind(unsigned Opcode) const = 0;
-  virtual const MCSlotInfo *getSlotInfo(const MCSlotKind Kind) const = 0;
+  virtual MCSlotKind getSlotKind(unsigned Opcode) const {
+    llvm_unreachable("Target didn't implement getSlotKind");
+  }
+  virtual const MCSlotInfo *getSlotInfo(const MCSlotKind Kind) const {
+    llvm_unreachable("Target didn't implement getSlotInfo");
+  }
   /// Return the Packet formats for this target
-  virtual const PacketFormats &getPacketFormats() const = 0;
+  virtual const PacketFormats &getPacketFormats() const {
+    llvm_unreachable("Target didn't implement getPacketFormats");
+  }
   /// Return a nop of the given byte size, or the smallest if zero.
-  virtual unsigned getNopOpcode(size_t Size = 0) const = 0;
+  virtual unsigned getNopOpcode(size_t Size = 0) const {
+    llvm_unreachable("Target didn't implement getNopOpcode");
+  }
   /// Return an opcode that reverses the branch condition of a given
   /// instruction
   /// \param Opc Opcode of the branch to reverse
   /// \pre Opc must be a conditional branch
-  virtual unsigned getOppositeBranchOpcode(unsigned Opc) const = 0;
+  virtual unsigned getOppositeBranchOpcode(unsigned Opc) const {
+    llvm_unreachable("Target didn't implement getOppositeBranchOpcode");
+  }
   /// Return the opcode of an unconditional jump
-  virtual unsigned getJumpOpcode() const = 0;
+  virtual unsigned getJumpOpcode() const {
+    llvm_unreachable("Target didn't implement getJumpOpcode");
+  }
   /// Return Multi-Slot Pseudo opcode based on Reg type and imm. size
   virtual unsigned getConstantMovOpcode(MachineRegisterInfo &MRI,
                                         unsigned int Reg, APInt &Val) const {
-    return -1;
+    llvm_unreachable("Target didn't implement getConstantMovOpcode");
```

Comment:
I really like this change. This removes some weird thinks, like some unsigned return functions that return negative values - this can hide some undesired behavior if the subclass miss the implementation.